### ISSUE TITLE
MQL query returning empty with QueryTimeSeriesRequest

### ISSUE THREAD

**Issue by RuiAAPeres** [googleapis/python-monitoring#492]
#### Environment details

  - OS type and version:
  - Python version: `3.10.5`
  - pip version: `22.1.2`
  - `google-cloud-monitoring` version: `2.11.1`

#### Steps to reproduce

I am trying to query something using MQL. I can see the values in the dashboard in GCP, but when querying via code, it returns `QueryTimeSeriesPager<time_series_descriptor {}`. I am using the three recommended metrics for Pub/Sub Push: `push_request_count`, `num_outstanding_messages` and `push_request_latencies`. The MQL looks like this:

```python
def make_request_count_query(value: str, value1: str) -> str:
    return f"""
            fetch pubsub_subscription
            | metric 'pubsub.googleapis.com/subscription/push_request_count'
            | filter
                (resource.subscription_id =~ '.*{value}*') &&
                (resource.subscription_id =~ '.*{value1}.*')
                | filter_ratio_by [], metric.response_class == 'ack'
"""


def make_outstanding_messages_query(value: str, value1: str) -> str:
    return f"""
            fetch pubsub_subscription
            | metric 'pubsub.googleapis.com/subscription/num_outstanding_messages'
            | filter
                (resource.subscription_id =~ '.*{value}.*')
                 && (resource.subscription_id =~ '.*{value1}.*')
            | max
"""


def make_latency_query(value: str, value1: str) -> str:
    return f"""
            fetch pubsub_subscription
            | metric 'pubsub.googleapis.com/subscription/push_request_latencies'
            | filter
                (resource.subscription_id =~ '.*{value}.*') &&
                (resource.subscription_id =~ '.*{value1}.*')
            | align delta(1m)
            | every 1m
            | group_by [resource.subscription_id],
                [value_push_request_latencies_percentile:
                percentile(value.push_request_latencies, 99)]
"""

```

I make the calls the following way:

```
client: monitoring_v3.QueryServiceClient =  ...
request = monitoring_v3.QueryTimeSeriesRequest(name=project_name, query=query)
result = client.query_time_series(request=request)
```

`result` comes as `QueryTimeSeriesPager<time_series_descriptor {}`. This happens for all three queries.

I can see the values properly populated in the dashboard, is this a bug?

**Comment by parthea** [comment]
Hi @RuiAAPeres,

I was able to re-create the behaviour that you described. This is intended behaviour. You'll need to iterate through the [QueryTimeSeriesResponse](https://cloud.google.com/python/docs/reference/monitoring/latest/google.cloud.monitoring_v3.types.QueryTimeSeriesResponse). The [response](https://cloud.google.com/python/docs/reference/monitoring/latest/google.cloud.monitoring_v3.types.QueryTimeSeriesResponse) is a pager is that you can loop through it to extract each result:

You'll find a [TimeSeriesDescriptor](https://cloud.google.com/python/docs/reference/monitoring/latest/google.cloud.monitoring_v3.types.TimeSeriesDescriptor) and [TimeSeriesData](https://cloud.google.com/python/docs/reference/monitoring/latest/google.cloud.monitoring_v3.types.TimeSeriesData)

```
client: monitoring_v3.QueryServiceClient =  ...
request = monitoring_v3.QueryTimeSeriesRequest(name=project_name, query=query)
result = client.query_time_series(request=request)
for page in result.pages:
    print(page.time_series_descriptor)
    for data in page.time_series_data:
        print(data)
        print(data.label_values)
        print(data.point_data)
```

**Comment by lvaylet** [comment]
Hi @parthea,

I was also able to reproduce the behaviour noticed by @RuiAAPeres. However, I am unable to extract any meaningful information from the pager.

Here is my code:

```python
from google.cloud import monitoring_v3

client: monitoring_v3.QueryServiceClient = monitoring_v3.QueryServiceClient()

request = monitoring_v3.QueryTimeSeriesRequest(
    name="projects/slo-generator-demo",
    query="""fetch gae_app
| metric 'appengine.googleapis.com/http/server/response_latencies'
| align delta(1m)
| every 1m
| group_by [],
    [value_response_latencies_percentile:
       percentile(value.response_latencies, 50)]"""
)

result = client.query_time_series(request)

print(result)

for page in result.pages:
    print(page.time_series_descriptor)
    for data in page.time_series_data:
        print(data)
        print(data.label_values)
        print(data.point_data)
```

And here is the output:

```bash
$ python mql_test.py 
QueryTimeSeriesPager<time_series_descriptor {
}
>

```

I expected more data from the `for` loop.

The query works as expected in the Cloud Monitoring console. I am also pretty sure the authentication is set up properly (through `gcloud auth application-default login`) and the project name is right. Otherwise I would get errors from the API server.

An extra pair of eyes is welcome. I might be missing something obvious.

**Comment by lvaylet** [comment]
Also note that the [official documentation for `query_time_series`](https://cloud.google.com/python/docs/reference/monitoring/latest/google.cloud.monitoring_v3.services.query_service.QueryServiceClient#google_cloud_monitoring_v3_services_query_service_QueryServiceClient_query_time_series) iterates over `result` itself instead of `result.pages`:

```python
# Make the request
    page_result = client.query_time_series(request=request)

    # Handle the response
    for response in page_result:
        print(response)
```

Not sure it is relevant though. I tried both approaches and they seem to give the same output, with no errors from the Python interpreter.

**Comment by lvaylet** [comment]
Hi @parthea,

I was indeed still experiencing issues until just a couple of hours ago. To be precise, I was getting an empty response with this code roughly 90% of the time. Then suddenly I got the response I was expecting after running the code again for the 10th or 11th time in a row. Then 10-15 empty results again. And so on. In the meantime, I was getting the expected results consistently when running the queries from the console. At this point, it struck me that the UI automatically sets the time window to 1 hour by default and this does not necessarily show up in the query itself. I solved my problem by including a `| within 1h` clause at the end of every query, so as to be explicit regarding the time window.

Anyway, thanks for helping me confirm that the code was right by allowing me to query your demo project. That helped me narrow down the possible causes!
